codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -69,409 +71,1060 @@
+                        // write(...) or writev(...) operation on the socket.
+                        if (inFireChannelReadComplete) {
+                            flushPending = true;
+                        } else {
+                            flush0(ctx);
+                        }
+                    } finally {
+                        writeDoneAndNoFlush = false;
+                    }
+                } else {
+                    // There is nothing to flush so this is a NOOP.
+                }
+            }
+
+            @Override
+            public ChannelPromise voidPromise() {
+                return unsafeVoidPromise;
+            }
+
+            @Override
+            public ChannelOutboundBuffer outboundBuffer() {
+                // Always return null as we not use the ChannelOutboundBuffer and not even support it.
+                return null;
+            }
         }
 
-        Http2StreamChannel channel() {
-            return childChannel;
+        /**
+         * {@link ChannelConfig} so that the high and low writebuffer watermarks can reflect the outbound flow control
+         * window, without having to create a new {@link WriteBufferWaterMark} object whenever the flow control window
+         * changes.
+         */
+        private final class Http2StreamChannelConfig extends DefaultChannelConfig {
+
+            Http2StreamChannelConfig(Channel channel) {
+                super(channel);
+                setRecvByteBufAllocator(new Http2StreamChannelRecvByteBufAllocator());
+            }
+
+            @Override
+            public int getWriteBufferHighWaterMark() {
+                return min(parent().config().getWriteBufferHighWaterMark(), initialOutboundStreamWindow);
+            }
+
+            @Override
+            public int getWriteBufferLowWaterMark() {
+                return min(parent().config().getWriteBufferLowWaterMark(), initialOutboundStreamWindow);
+            }
+
+            @Override
+            public MessageSizeEstimator getMessageSizeEstimator() {
+                return FlowControlledFrameSizeEstimator.INSTANCE;
+            }
+
+            @Override
+            public WriteBufferWaterMark getWriteBufferWaterMark() {
+                int mark = getWriteBufferHighWaterMark();
+                return new WriteBufferWaterMark(mark, mark);
+            }
+
+            @Override
+            public ChannelConfig setMessageSizeEstimator(MessageSizeEstimator estimator) {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            @Deprecated
+            public ChannelConfig setWriteBufferHighWaterMark(int writeBufferHighWaterMark) {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            @Deprecated
+            public ChannelConfig setWriteBufferLowWaterMark(int writeBufferLowWaterMark) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the missing @deprecated Javadoc tag. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)